services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainController.java

```diff
@@ -501,7 +502,8 @@ private void onStoreInitialized() {
       currentSlot = deltaTime.dividedBy(UnsignedLong.valueOf(SECONDS_PER_SLOT));
     } else {
       UnsignedLong timeUntilGenesis = genesisTime.minus(currentTime);
```

Comment:
The `if` case should actually be `if (currentTime.compareTo(genesisTime) >= 0) {` so that if we execute this on the exact genesis time we should start the chain.

Reply:
fair call, i can update that while i'm here.